src/include/function/cast/functions/cast_string_to_functions.h

```diff
@@ -0,0 +1,269 @@
+#pragma once
+
+#include <cassert>
+
+#include "cast_string_non_nested_functions.h"
+#include "common/copier_config/copier_config.h"
+#include "common/exception/runtime.h"
+#include "common/string_format.h"
+#include "common/type_utils.h"
+#include "common/types/blob.h"
+#include "common/vector/value_vector.h"
+
+namespace kuzu {
+namespace function {
+
+struct CastStringToTypes {
+    // non-nested types: used externally
+    template<typename T>
+    static inline void operation(
+        common::ku_string_t& input, T& result, common::ValueVector& resultVector) {
+        // base case: int64
+        simpleIntegerCast<T, true>((char*)input.getData(), input.len, result,
+            common::LogicalType{common::LogicalTypeID::INT64});
+    }
+
+    template<typename T>
+    static inline bool tryCast(const char* input, uint64_t len, T& result) {
+        // try cast for signed integer types
+        return trySimpleIntegerCast<T, true>(input, len, result);
+    }
+
+    // non nested-types (const char* are used internally)
+    template<typename T>
+    static inline void operation(const char* input, uint64_t len, T& result) {
+        // base case: int64
+        simpleIntegerCast<int64_t>(
+            input, len, result, common::LogicalType{common::LogicalTypeID::INT64});
+    }
+
+    // nested types
+    template<typename T>
+    static void operation(const char* input, uint64_t len, common::ValueVector* vector,
```

Comment:
Our ultimate goal is to only keep one string cast operation. CSVReaderConfig should be passed as bindData, and rowToAdd should be replaced with dstValue. You can take a look at how duckdb achieves this, they have a class called BoundCastInfo. You can change this in later PRs.